### PR TITLE
fix: `det user change-password` for user's own passwords.

### PR DESCRIPTION
**Pull request by ioga**
## Description

`det user change-password` for changing your own password was broken because of auth.

## Test Plan

run `det user change-password` to change your own password.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: harness/determined/cli/user.py [hunk 1]
```diff
@@ -78,7 +78,7 @@ def change_password(parsed_args: Namespace) -> None:
     elif parsed_args.user:
         username = parsed_args.user
     else:
-        username = authentication.must_cli_auth().get_session_user()
+        username = client.get_session_username()
```
Comment (NicholasBlaskey) [inline harness/determined/cli/user.py:81]: ```must_cli_auth``` raising an exception for a CLI using auth seems rather unexpected

Comment (NicholasBlaskey) [inline harness/determined/cli/user.py:81]: Release note too? Not sure when this broke

**Review by NicholasBlaskey — APPROVED** [review]
LGTM

File: harness/determined/cli/user.py (hunk 1 above)
Comment (ioga) [reply harness/determined/cli/user.py:81]: python sdk has one singleton, cli has another. ideally we'd move to the one for python sdk completely...

**Pull request merged**